File: RangeRev/Range.scala
/*
    Author : Pakshal Jain
    Date : 12/12/2022
    Program : To Display Entered range in Reverse Order
*/

import scala.io.StdIn.*;

//Class Declaration with Constructor of Class
class RangeRev(No : Int,No2 : Int) {
    //Characteristics of Class
    var i : Int = 0;

    //Behaviour of Class
    def Display() : Unit = {
        println("Printing Numbers in Entered Range in Reverse Order");
        for(i <- this.No2 to this.No by -1) {
            print(i+"\t");
        }
    }
}

//Main object 
object Range {
    //Main Function
    def main(args : Array[String]) : Unit = {
        
        var No : Int = 0;
        var No2 : Int = 0;

        println("Jay ganesh...");

        //Taking Input From User
        println("Enter Starting Number");
        No = readInt();

        println("Enter Ending Number");
        No2 = readInt();

        //Handeling Condition if Entered Number is Negative or Starting Number is Greter Than Ending Number

        if((No < 0)||(No > No2))     
        {
            println("Invalid Range");
            return;
        }

        //Creating Object of RangeRev Class
        var robj = new RangeRev(No,No2);
        //Call To Behaviour of Class
        robj.Display();
    }
}

File: RangeRev/Range.go
/*
	Author : Pakshal jain
	Date : 12/12/2022
	Program : To Display Entered Range in Reverse Order
*/

//Required Libraries
package main
import "fmt"

///////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////

//Function To Print Entered Range in Reverse Order
func Range(No int,No2 int) {
	var i int = 0;

	fmt.Println("Printing Numbers of Entered Range in Reverse Order");
	for i = No2;i >= No;i-- {
		fmt.Print(i,"\t");
	}
}

/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////

//Main Function
func main() {
	var No int = 0;
	var No2 int = 0;

	fmt.Println("Jay Ganesh..");

	//Taking Input From User
	fmt.Println("Enter Starting Number");
	fmt.Scan(&No);

	fmt.Println("Enter Ending Number");
	fmt.Scan(&No2);

	//Handeling Condition if Number is Negative or Starting Number is Greter than Ending Number
	if No < 0 || No > No2 {
		fmt.Println("Invalid Range");

		return;
	}

	//Call To Range Function
	Range(No,No2);
}